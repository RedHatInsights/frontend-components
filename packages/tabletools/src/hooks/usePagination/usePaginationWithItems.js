import usePagination from './usePagination';

const usePaginationWithItems = (options) => {
  const enablePagination = options?.pagination !== false;
  const paginate = usePagination(options);

  const paginator = (items) => {
    const { page, perPage } = paginate.pagination;
    const start = (page - 1) * perPage;
    const end = start + perPage;

    return items.slice(start, end);
  };

  return enablePagination
    ? {
        ...paginate,
        paginator,
      }
    : {};
};

export default usePaginationWithItems;
