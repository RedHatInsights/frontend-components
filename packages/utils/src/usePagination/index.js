export { default } from './usePagination';
export { default as usePaginationWithItems } from './usePaginationWithItems';
