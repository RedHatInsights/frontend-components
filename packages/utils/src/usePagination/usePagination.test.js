import { act, renderHook } from '@testing-library/react-hooks';
import usePagination from './usePagination';

describe('usePagination', () => {
  it('returns a paginate configuration', () => {
    const { result } = renderHook(() => usePagination());
    expect(result).toMatchSnapshot();
  });

  it('returns a paginate configuration', () => {
    const { result } = renderHook(() => usePagination());

    act(() => {
      result.current.setPage(2);
    });

    expect(result.current.toolbarProps.pagination.page).toBe(2);
  });
});
