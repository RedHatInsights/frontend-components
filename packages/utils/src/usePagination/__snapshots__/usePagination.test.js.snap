// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePagination returns a paginate configuration 1`] = `
Object {
  "all": Array [
    Object {
      "pagination": Object {
        "page": 1,
        "perPage": 10,
      },
      "setPage": [Function],
      "toolbarProps": Object {
        "pagination": Object {
          "onPerPageSelect": [Function],
          "onSetPage": [Function],
          "page": 1,
          "perPage": 10,
        },
      },
    },
  ],
  "current": Object {
    "pagination": Object {
      "page": 1,
      "perPage": 10,
    },
    "setPage": [Function],
    "toolbarProps": Object {
      "pagination": Object {
        "onPerPageSelect": [Function],
        "onSetPage": [Function],
        "page": 1,
        "perPage": 10,
      },
    },
  },
  "error": undefined,
}
`;
